data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/handler/ContractAgreementHandler.java

```diff
@@ -105,19 +104,18 @@ public boolean canHandle(@NotNull MultipartRequest multipartRequest) {
                 .asset(asset)
                 .build();
 
-        // Create contract agreement
-        Result<ContractAgreement> result = transformerRegistry.transform(input, ContractAgreement.class);
+        var result = transformerRegistry.transform(input, ContractAgreementTransformerOutput.class);
         if (result.failed()) {
             monitor.debug(String.format("Could not transform contract agreement: [%s]",
                     String.join(", ", result.getFailureMessages())));
             return createBadParametersErrorMultipartResponse(message);
         }
 
         // TODO get hash from message
-        var agreement = result.getContent();
+        var output = result.getContent();
         var processId = message.getTransferContract();
         var negotiationResponse = negotiationManager.confirmed(claimToken,
-                String.valueOf(processId), agreement, null);
+                String.valueOf(processId), output.getContractAgreement(), null);
```

Comment:
The policy from the output should be persisted. Just the agreement is handed over to the negotiation manager, so the policy is lost at that point. Therefore, the agreement then references a policy id that does not exist in the store.

Reply:
@ronjaquensel thanks, yes, this made total sense, I was just wondering why nobody was using that policy object but it's because it needs to be stored locally so then after it can be used later on. I will introduce policy persistence in the `confirmed` method of the negotiation manager.